src/Models/AdvectedPopulations/PISCES/dissolved_organic_matter.jl

```diff
@@ -0,0 +1,157 @@
+
+#The carbon compartment of the model is composed of P, D, Z, M, DOC, POC, GOC, DIC and CaCO₃. Carbon is conserved within the model.
+#These pools of carbon have complex interactions.
+#Particles of carbon may degrade into smaller particles, or aggregate into larger particles. 
+#Phytoplankton uptake DIC for biological processes including photosynthese. 
+#Mortality returns carbon in biomasses to particles of carbon.
+#Remineralisation processes break organic carbon down into inorganic carbon.
+#Particles of carbon experience sinking, and this is significant in tracking carbon export to the deep ocean.
+
+#This document contains forcing equations for:
+    #P_up, R_up (eqs30a, 30b) 
+    #Remin, Denit (eqs 33a, 33b)
+    #Bacteria population (eq35)
+    #Aggregation of DOC (eq36)
+    #Degradation rate of POM (applies to POC and GOC, eq38)
+    #Limiting nutrients for bacteris (eq34)
+    #Forcing for DOC
+
+#Remineralisation of DOM can be either oxic (Remin), or anoxic (Denit). This is regulated by oxygen_concentration. 
+#Remineralisation processes are responsible for breaking down organic matter back into inorganic forms. These terms contribute to forcing equations for inorganic nutrients.
+#Remineralisation occurs in oxic waters.
+@inline function oxic_remineralization(O₂, NO₃, PO₄, NH₄, DOC, T, bFe, Bact, bgc)
+    O₂ᵘᵗ = bgc.OC_for_ammonium_based_processes
+    λ_DOC = bgc.remineralisation_rate_of_DOC
+    bₚ = bgc.temperature_sensitivity_of_growth
+    Bactᵣₑ = bgc.bacterial_reference
+
+    Lₗᵢₘᵇᵃᶜᵗ = bacterial_activity(DOC, PO₄, NO₃, NH₄, bFe, bgc)[2]
+
+    #min((O₂)/(O₂ᵘᵗ), λ_DOC*bₚ^T*(1 - oxygen_conditions(O₂, bgc)) * Lₗᵢₘᵇᵃᶜᵗ * (Bact)/(Bactᵣₑ) * DOC), definition from paper did not make sense with dimensions, modified below
+    return λ_DOC*bₚ^T*(1 - oxygen_conditions(O₂, bgc)) * Lₗᵢₘᵇᵃᶜᵗ * (Bact)/(Bactᵣₑ + eps(0.0)) * DOC #33a
+end
+
+#Denitrification is the remineralisation process in anoxic waters.
+@inline function denitrification(NO₃, PO₄, NH₄, DOC, O₂, T, bFe, Bact, bgc)
+    λ_DOC = bgc.remineralisation_rate_of_DOC
+    rₙₒ₃¹ = bgc.CN_ratio_of_denitrification
+    bₚ = bgc.temperature_sensitivity_of_growth
+    Bactᵣₑ = bgc.bacterial_reference
+
+    Lₗᵢₘᵇᵃᶜᵗ = bacterial_activity(DOC, PO₄, NO₃, NH₄, bFe, bgc)[2]
+
+    #min(NO₃/rₙₒ₃¹, λ_DOC*bₚ^T* oxygen_conditions(O₂, bgc)* Lₗᵢₘᵇᵃᶜᵗ*(Bact)/(Bactᵣₑ) * DOC), definition from paper did not make sense with dimensions, modified below
+    return λ_DOC*bₚ^T* oxygen_conditions(O₂, bgc)* Lₗᵢₘᵇᵃᶜᵗ*(Bact)/(Bactᵣₑ + eps(0.0)) * DOC #33b
+end
+
+#Bacteria are responsible for carrying out biological remineralisation processes. They are represent in the following formulation, with biomass decreasing at depth.
+@inline bacterial_biomass(zₘₐₓ, z, Z, M) = ifelse(abs(z) <= zₘₐₓ, min(0.7*(Z + 2*M), 4), min(0.7*(Z + 2*M), 4)*(abs(zₘₐₓ/(z + eps(0.0)))^0.683))  #35b
```

Comment:
To-do: make this just use the non-abs version because its confusing that we're passing around z_max positive and z negative